src/library_egl.js

```diff
@@ -15,6 +15,28 @@ var LibraryEGL = {
     currentContext: 0 /* EGL_NO_CONTEXT */,
     currentReadSurface: 0 /* EGL_NO_SURFACE */,
     currentDrawSurface: 0 /* EGL_NO_SURFACE */,
+    // Because there's only a single global EGLContext with a single EGLSurface,
+    // we have to store the state for what EGLConfig is being used here. 
+    currentEGLConfig: 0,
+    // List of different EGLConfig configurations in spec order
+    configAttribs: [
+        {alpha: 8,  samples: 0, buffers: 0,  depth: 0,   stencil: 0},
```

Comment:
Perhaps this would be better to be a nested set of for loops, that way in the future the code size won't explode as new configs are added. E.g.

    for(var alpha of [0, 8]) {
      for(var samples of [0, 4]) {
        for( ... ) {
          configs.push({alpha: alpha, samples: samples, ... });
        }
      }
    }

Skimming through the EGL spec, the caps `EGL_ALPHA_SIZE`, `EGL_CONFIG_CAVEAT`, `EGL_DEPTH_SIZE`, `EGL_STENCIL_SIZE`, `EGL_SAMPLE_BUFFERS`, `EGL_SAMPLES`, `EGL_MIN_SWAP_INTERVAL`, `EGL_MAX_SWAP_INTERVAL` and `EGL_TRANSPARENT_TYPE` are items that have a meaning for WebGL as well, and even if we don't add all of them right now, those could be added later, so good to have a compact representation that is easily extended.